### PR TITLE
fix: don't use ! as a delimiter in the store

### LINKED ISSUE #276: `!` used as delimiter in store conflicts with webpack
Webpack uses `!` as delimiter between loaders and resources. That means that loading packages from store messes with that. It would be great if pnpm changes `!` to `___` for example. What do you think?

### PR DESCRIPTION
**Pull request by zkochan**
! can be part of a valid npm package name. Use + as a delimiter instead.

close #276

BREAKING CHANGE:

Stores created with the ! delimiter are not compatible with the new version that uses +.

Any store created by older versions of pnpm has to be removed and reinstalled.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "pnpm",
   "description": "A fast implementation of npm install",
-  "version": "0.26.2",
+  "version": "0.27.0",
```
Comment (zkochan) [inline package.json]: I had to update it, otherwise the tests wouldn't pass

File: lib/pkg_full_name.js [hunk 2]
```diff
@@ -1,2 +1,2 @@
 'use strict'
-module.exports = pkg => pkg.name.replace('/', '!') + '@' + pkg.version
+module.exports = pkg => pkg.name.replace('/', '+') + '@' + pkg.version
```
Comment (andreypopp) [inline lib/pkg_full_name.js]: Might be worth a separate utility function.

Comment (zkochan) [reply lib/pkg_full_name.js]: good idea, I'll do that

**Comment by andreypopp** [comment]
What's the ETA for this fix? Can I help somehow?

**Comment by andreypopp** [comment]
Tested it on a project which has deps hosted on a github: webpack failures gone.

**Comment by zkochan** [comment]
Published in v0.28.0

**Comment by andreypopp** [comment]
Thanks!

**Pull request merged**